carbons/carbons.go

```diff
@@ -110,3 +111,40 @@ func Unwrap(del *delay.Delay, r xml.TokenReader) (xml.TokenReader, xml.StartElem
 	out, err := forward.Unwrap(del, xmlstream.Inner(r))
 	return out, se, err
 }
+
+// Private is an xmlstream.Transformer that excludes a <message/> from being forwarded to
+// other Carbons-enabled resources, by adding a <private/> element and a <no-copy/> hint.
+// The first call to consume the output stream will return an error if the input stream
+// is not a message element with an appropriate namespace.
+func Private(r xml.TokenReader) xml.TokenReader {
+	token, err := r.Token()
+	if err != nil && err != io.EOF {
+		return xmlstream.ReaderFunc(func() (xml.Token, error) {
+			return token, err
+		})
+	}
+	se, ok := token.(xml.StartElement)
```

Comment:
Alright, I tried to change the implementation in order to deactivate cc'ing for all top-level messages.

Per your recommendations, I used the `xmlstream.InsertFunc`. To keep things simple, I add the `<private/>`/`<no-copy/>` to the start of the message instead of the end (I would need to consume the `EndElement` to know I'm reaching the end, but doing so messes with the level counter in `InsertFunc`).